ts/src/xt.ts

```diff
@@ -0,0 +1,4094 @@
+        const marketId = this.safeString (position, 'symbol');
+        market = this.safeMarket (marketId, market, undefined, 'contract');
+        const symbol = this.safeSymbol (marketId, market, undefined, 'contract');
+        const positionType = this.safeString (position, 'positionType');
+        const marginMode = (positionType === 'CROSSED') ? 'cross' : 'isolated';
+        const collateral = this.safeNumber (position, 'isolatedMargin');
+        return this.safePosition ({
+            'info': position,
+            'id': undefined,
+            'symbol': symbol,
+            'timestamp': undefined,
+            'datetime': undefined,
+            'hedged': undefined,
+            'side': this.safeStringLower (position, 'positionSide'),
+            'contracts': this.safeNumber (position, 'positionSize'),
+            'contractSize': market['contractSize'],
+            'entryPrice': this.safeNumber (position, 'entryPrice'),
+            'markPrice': undefined,
+            'notional': undefined,
+            'leverage': this.safeInteger (position, 'leverage'),
+            'collateral': collateral,
+            'initialMargin': collateral,
+            'maintenanceMargin': undefined,
+            'initialMarginPercentage': undefined,
+            'maintenanceMarginPercentage': undefined,
+            'unrealizedPnl': undefined,
+            'liquidationPrice': undefined,
+            'marginMode': marginMode,
+            'percentage': undefined,
+            'marginRatio': undefined,
+        });
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
+        //
+        // spot: error
+        //
+        //     {
+        //         "rc": 1,
+        //         "mc": "AUTH_103",
+        //         "ma": [],
+        //         "result": null
+        //     }
+        //
+        // spot: success
+        //
+        //     {
+        //         "returnCode": 0,
+        //         "msgInfo": "success",
+        //         "error": null,
+        //         "result": []
+        //     }
+        //
+        // swap and future: error
+        //
+        //     {
+        //         "returnCode": 1,
+        //         "msgInfo": "failure",
+        //         "error": {
+        //             "code": "403",
+        //             "msg": "invalid signature"
+        //         },
+        //         "result": null
+        //     }
+        //
+        // swap and future: success
+        //
+        //     {
+        //         "returnCode": 0,
+        //         "msgInfo": "success",
+        //         "error": null,
+        //         "result": null
+        //     }
+        //
+        // other:
+        //
+        //     {
+        //         "rc": 0,
+        //         "mc": "SUCCESS",
+        //         "ma": [],
+        //         "result": {}
+        //     }
+        //
+        const status = this.safeStringUpper2 (response, 'msgInfo', 'mc');
```

Comment:
I think we should try to process messages like these ⬇️  as an error response
```

handleRestResponse:
 xt GET https://fapi.xt.com/future/trade/v1/order/detail 200 OK 
ResponseHeaders:
 {
  'Cf-Cache-Status': 'DYNAMIC',
  'Cf-Ray': '7b41b062bb34dc45-LHR',
  Connection: 'keep-alive',
  'Content-Encoding': 'gzip',
  'Content-Type': 'application/json',
  Date: 'Fri, 07 Apr 2023 10:46:50 GMT',
  Server: 'cloudflare',
  'Set-Cookie': '__cf_bm=8GAAHtf21yq_.MbtqKcElh2qOscOkkxymDoKRoyb1Ro-1680864410-0-AWmagr3uX3m1cnyQ6lHh9ChXADwja2j0xviOnIEDCpY5QwUw5qgkcs0vDoco9VWov++A+vWGV20ltCvZjyaqopg=; path=/; expires=Fri, 07-Apr-23 11:16:50 GMT; domain=.xt.com; HttpOnly; Secure; SameSite=None',
  'Transfer-Encoding': 'chunked',
  'X-Ratelimit-Burst-Capacity': '200',
  'X-Ratelimit-Remaining': '199',
  'X-Ratelimit-Replenish-Rate': '200',
  'X-Ratelimit-Requested-Tokens': '1'
} 
ResponseBody:
 {"returnCode":0,"msgInfo":"success","error":null,"result":null} 

2023-04-07T10:46:50.640Z iteration 0 passed in 860 ms

{
  info: {},
  id: undefined,
  clientOrderId: undefined,
  timestamp: undefined,
  datetime: undefined,
  lastTradeTimestamp: undefined,
  symbol: 'LTC/USDT:USDT',
  type: undefined,
  timeInForce: undefined,
```

Reply:
Some methods return that as a successful response though like setLeverage, addMargin, reduceMargin

Reply:
@Dan-krm you're right!